tools/src/main/java/org/apache/kafka/trogdor/agent/WorkerManager.java

```diff
@@ -250,6 +250,10 @@ synchronized void waitForQuiescence() throws InterruptedException {
             this.reference = shutdownManager.takeReference();
         }
 
+        boolean hasExpired() {
```

Comment:
I don't know if it makes sense to add this function just for a single caller.

We could just check `if (spec.endMs() >= time.milliseconds())` below...